pkg/apis/validation/validation.go

```diff
@@ -51,6 +56,24 @@ func ValidateAttributes(attr map[string]string) error {
 	return nil
 }
 
+func keyUsages(ss string, errs []string) []string {
+	if len(ss) == 0 {
+		return errs
+	}
+
+	usages := strings.Split(ss, ",")
+
+	for _, usage := range usages {
+		if _, ok := cmapiutil.ExtKeyUsageType(cmapi.KeyUsage(usage)); !ok {
+			if _, ok := cmapiutil.KeyUsageType(cmapi.KeyUsage(usage)); !ok {
+				errs = append(errs, fmt.Sprintf("%q is not a valid key usage", usage))
+			}
+		}
+	}
```

Comment:
Maybe check for duplicate usages and report an error, or de-duplicate....perhaps that happens in mutating webhook?....I'll keep reading...